engine/access/rest/rest_api.go

```diff
@@ -0,0 +1,112 @@
+package rest
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const MaxBlockIDCount = 50
+
+// RestAPIHandler provides the implementation of each of the REST API
+type APIHandler struct {
+	backend *backend.Backend
+	logger  zerolog.Logger
+}
+
+func NewRestAPIHandler(backend *backend.Backend, logger zerolog.Logger) *APIHandler {
+	return &APIHandler{
+		backend: backend,
+		logger:  logger,
+	}
+}
+
+func (restAPI *APIHandler) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	// create a logger for the request
+	errorLogger := restAPI.logger.With().Str("request_url", r.URL.String()).Logger()
+
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+
+	vars := mux.Vars(r)
+	idParam := vars["id"]
+
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
+
+	ids := strings.Split(idParam, ",")
+
+	blocks := make([]*generated.Block, len(ids))
+
+	if len(ids) > MaxBlockIDCount {
+		restAPI.errorResponse(w, http.StatusBadRequest, fmt.Sprintf("at most %d Block IDs can be requested at a time", MaxBlockIDCount), errorLogger)
+		return
+	}
+
+	for i, id := range ids {
+		flowID, err := flow.HexStringToIdentifier(id)
```

Comment:
we should validate `id` before parsing it. [HexStringToIdentifier](https://github.com/onflow/flow-go/blob/master/model/flow/identifier.go#L36-L36) will decode the input first, then return an error if it's not valid. This means that an attacker could attempt to exploit the `hex.Decode` method with malicious ids. e.g. send a list of very large IDs to tie up memory, or exploit some undiscovered bug in the implementation.

we should check that the ID is the correct length and only hex characters. there are a number of libraries we could use to simplify this for all of the inputs we're going to receive in this api.

Reply:
It might even be good to extract that into a parser for IDs reused whenever we parse IDs as part of the API so there is no regression.

Reply:
I started adding this but then realized as @sideninja suggested that this needs to be its own validation layer. Hence, I created this [issue](https://github.com/dapperlabs/flow-go/issues/6006) to tackle is separately.

Reply:
I'm ok tracking that as a separate issue if we pick that issue up before implementing the rest of the endpoints.